File: test-workspaces/modules/shared_common/src/utility_aes404_complex_parser.py
# AES404: utility with complex logic (should be pure function)
def parse_complex(input_str):
    if not input_str:
        raise ValueError("empty input")
    
    result = ""
    for ch in input_str:
        if ch.isalnum():
            result += ch
        elif ch == ' ':
            result += '_'
    
    if len(result) > 100:
        raise ValueError("too long")
    
    return result

File: test-workspaces/crates/shared_common/src/utility_aes404_complex_parser.rs
// AES404: utility with complex logic (should be pure function)
pub fn parse_complex(input: &str) -> Result<String, String> {
    if input.is_empty() {
        return Err("empty input".to_string());
    }
    
    let mut result = String::new();
    for ch in input.chars() {
        if ch.is_alphanumeric() {
            result.push(ch);
        } else if ch == ' ' {
            result.push('_');
        }
    }
    
    if result.len() > 100 {
        return Err("too long".to_string());
    }
    
    Ok(result)
}
